### PR TITLE
Fix package file list cache

### PR DESCRIPTION
**Pull request by ChrisBr**
We need to use a custom cache key including the locals otherwise the cache will be stale.

Follow up of https://github.com/openSUSE/open-build-service/pull/6590

image

### CODE REVIEW

File: src/api/app/views/webui/package/_files_view.html.erb [hunk 1]
```diff
@@ -9,7 +9,12 @@
       </tr>
       </thead>
       <tbody>
-        <%= render partial: 'file', collection: @files, cached: true, locals: { project: @project, package: @package, expand: @expand } %>
+        <%- file_locals = {
+          project: @project, package: @package, expand: @expand, is_current_rev: @is_current_rev,
+          can_modify: User.current.can_modify?(@package), nobody: User.current.is_nobody?
+        } %>
+        <%= render partial: 'file', collection: @files, cached: proc { |file| [file, file_locals].hash }, locals: file_locals
+        %>
```
Comment (coolo) [inline src/api/app/views/webui/package/_files_view.html.erb]: about time the bento pages for packages and projects find their rest :(

Comment (ChrisBr) [reply src/api/app/views/webui/package/_files_view.html.erb]: What do you mean?

Comment (coolo) [reply src/api/app/views/webui/package/_files_view.html.erb]: the whole theory behind "let's keep bento and bootstrap alive in parallel for a while" was based on the assumption that every developer can convert one action a day. As this assumption failed, my recommendation would be to give up and make project and package pages bootstrap only (risking a mixed environment for everyone) - and continue with beta flag only for pages under sprint development.

But that's unrelated to this PR - it just makes every fix in views a little more complicated

**Comment by bgeuken** [comment]
@ChrisBr Sorry, I saw this one to late. Though our packages are anyway in 'unresolvable' state right now.

Deployed state right now is 8536dcc77096dc55fd570ef14989830a40760df1

File: src/api/app/views/webui/package/_file.html.erb [hunk 2]
```diff
@@ -11,12 +11,12 @@
   <td><span class="hidden"><%= file[:mtime] %></span><%= timeago_tag Time.at(file[:mtime].to_i) %>
   </td>
   <!-- limit download for anonymous user to avoid getting killed by crawlers -->
-  <td><%= if !User.current.is_nobody? || file[:size].to_i < (4 * 1024 * 1024)
+  <td><%= if !nobody || file[:size].to_i < (4 * 1024 * 1024)
```
[the remark this answers is not in the document]
Comment (ChrisBr) [reply src/api/app/views/webui/package/_file.html.erb]: We can do that in a follow up as it is unrelated to this change!

**Pull request merged**